### PR TITLE
Continuous integration using GitHub Actions.

### PR DESCRIPTION
**Pull request by chrisdembia**
This PR introduces continuous integration for all 3 platforms using GitHub Actions.

This change is [Reviewable](https://reviewable.io/reviews/simbody/simbody/698)

### CODE REVIEW

File: SimTKmath/Optimizers/src/IpOpt/IpTNLPAdapter.cpp [hunk 1]
```diff
@@ -292,7 +292,7 @@ namespace SimTKIpopt
           }
         }
         else if (lower_bound > upper_bound) {
-          char string[128];
+          char string[129];
```
Comment (chrisdembia) [inline SimTKmath/Optimizers/src/IpOpt/IpTNLPAdapter.cpp:295]: @sherm1 this change gets rid of a GCC7 compiler error (#694). Are you okay with this change (and the one below)?

**Comment by chrisdembia** [comment]
Ready for review

**Comment by aymanhab** [comment]
@chrisdembia did you push a change? I see your comments but can't find the changes you made. Maybe I'm looking at the wrong version? Thanks

**Comment by chrisdembia** [comment]
Oops; I pushed the changes just now.

**Comment by chrisdembia** [comment]
Thanks @aymanhab and @sherm1 .

Merging.

**Pull request merged**